### PR TITLE
feat(sdk): support serialize and deserialize in PipelineHandler

### PR DESCRIPTION
**Pull request by jialeicui**
## Description
support serialize and deserialize in PipelineHandler
## Modules
- [ ] UI
- [ ] Controller
- [ ] Agent
- [x] Client
- [x] Python-SDK
- [ ] Others

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/starwhale/api/_impl/model.py [hunk 1]
```diff
@@ -288,36 +304,36 @@ def _starwhale_internal_run_ppl(self) -> None:
...
+        *args,
     ) -> None:
         self._status_writer.write(
             {
                 "time": now_str(),  # type: ignore
                 "status": exception is None,
                 "exception": str(exception),
                 "index": data.idx,
-                "output_size": len(pred),
+                "output_size": len(args[0]),
```
Comment (tianweidut) [inline client/starwhale/api/_impl/model.py:322]: if `args` is empty, `args[0]` will raise IndexError ?

**Review by tianweidut — COMMENTED** [review]
Other LGTM

**Comment by tianweidut** [comment]
#611

File: client/starwhale/api/_impl/model.py (hunk 1 above)
Comment (jialeicui) [reply client/starwhale/api/_impl/model.py:322]: can we assume that `ppl`  function will always return something for `cmp`?

Comment (tianweidut) [reply client/starwhale/api/_impl/model.py:322]: > can we assume that `ppl` function will always return something for `cmp`?

yes.

**Review by anda-ren — COMMENTED** [review]
LGTM

**Review by anda-ren — APPROVED** [review]
lgtm

**Review by tianweidut — APPROVED** [review]
LGTM

**Pull request merged**